### PR TITLE
Blacklight Version 7 Compatibility

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Comment by jcoyne** [comment]
@cbeer this is blocked by https://github.com/projectblacklight/blacklight/blob/master/app/controllers/concerns/blacklight/bookmarks.rb#L43
we need that `fetch` to take `fl: '*'` as it used to do that by default.   Ideas of how we should fix this?

**Comment by cbeer** [comment]
Because you need the `marc_display` field? Can we just add it to either the solr request handler configuration or catalog controller?

**Comment by jcoyne** [comment]
Yes, the document needs to have the `marc_display` field.  We *could* add it to the request handler, but that seems fragile to me.  Are you thinking we should add it to `default_solr_params` in `CatalogController`?

**Comment by cbeer** [comment]
`default_solr_params` sounds fine to me; I only suggested the request handler because that's where we're already setting the `fl`: https://github.com/projectblacklight/blacklight/blob/master/solr/conf/solrconfig.xml#L190 (and I'm not sure I buy the comment about it being /that/ expensive..)

**Comment by jcoyne** [comment]
Which comment about expensive?

**Comment by jcoyne** [comment]
I see the comment in solr config.xml

File: lib/generators/blacklight/marc/install_generator.rb [hunk 1]
```diff
@@ -56,11 +56,26 @@ def add_marc_extension_to_solrdocument
...
+
+    def add_default_fl_field
+      path = File.expand_path('app/controllers/catalog_controller.rb', destination_root)
+      say_status :gsub, relative_to_original_destination_root(path)
+      content = File.binread(path)
+      content.gsub!(/config\.default_solr_params\s+=\s+{([^}]*)}/) do
+                    "config.default_solr_params = {\n" \
+                    "      #{Regexp.last_match[1].strip},\n" \
+                    "      # Ensure that we fetch the needed fields when fetching the document\n" \
+                    "      fl: SolrDocument.extension_parameters[:marc_source_field]\n" \
```
Comment (cbeer) [inline lib/generators/blacklight/marc/install_generator.rb]: Isn't this saying to only return the `:marc_source_field`?

**Comment by cbeer** [comment]
This PR adds `marc_display` to the default solr config:

https://github.com/projectblacklight/blacklight/pull/1747

**Pull request merged**